### PR TITLE
Dockerfile used in k8s

### PR DESCRIPTION
**Pull request by wjxiz1992**
Signed-off-by: Allen Xu <allxu@nvidia.com>

Dockerfile used to run GPU accelerated Spark queries.
This PR adds in a dockerfile whose output image can be used in AWS EKS and GCS GKE environment.
There is also an example test python script to show the ability to:
1. leverage spark-rapids plugin for acceleration
2. show s3 storage accessibility
3. Spark UI access
4. Spark History Server access

This PR also include a README doc to show step by step replication for usage.

### CODE REVIEW

File: k8s/README.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+eksctl create cluster \
+--name test-cluster \
+--region us-west-2 \
+--with-oidc \
+--ssh-access \
+--ssh-public-key <user-key-pair> \
+--managed \
+--nodegroup-name gpu-node-group \
+--node-type g4dn.xlarge \
+--nodes 2
```
Comment (GaryShen2008) [inline k8s/README.md]: Missing a '\'.

File: k8s/README.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+   | AAAAAAAAABAAAAAA|
+   | AAAAAAAAABAAAAAA|
+   | AAAAAAAADBAAAAAA|
+   | AAAAAAAAEBAAAAAA|
+   +-----------------+
+   only showing top 20 rows
+   ```
+4. Delete the pod
+   ```bash
+   kubeclt delete pod example-driver
```
Comment (GaryShen2008) [inline k8s/README.md:181]: Shall we explain how to delete the node group and the cluster by the command?

Comment (wjxiz1992) [reply k8s/README.md:181]: I'm not sure, we assume the user knows how to manage their cluster or not?

File: k8s/README.md (hunk 1 above)
Comment (wjxiz1992) [reply k8s/README.md]: Added.

File: k8s/README.md (hunk 2 above)
Comment (wjxiz1992) [reply k8s/README.md:181]: Added

File: k8s/README.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+   CoreDNS is running at https://8440EE5F8730EDD7D8B989F704DE1DFE.gr7.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+   ```
+   The master IP address is `https://8440EE5F8730EDD7D8B989F704DE1DFE.gr7.us-west-2.eks.amazonaws.com`, it    will be the k8s master address when submitting Spark applications.
+
+2. create ServiceAccount and ClusterRoleBinding for the Spark application
+   ```bash
+   # service account named spark
+   kubectl create serviceaccount spark
+   # cluster role binding for the service account
+   kubectl create clusterrolebinding spark-role --clusterrole=edit --serviceaccount=default:spark    --namespace=default
```
Comment (GaryShen2008) [inline k8s/README.md]:
```suggestion
   kubectl create clusterrolebinding spark-role --clusterrole=edit --serviceaccount=default:spark --namespace=default
```

File: k8s/read-s3-test.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Import SparkSession
+from pyspark.sql import SparkSession
+import time
+
+# Create SparkSession 
+spark = SparkSession.builder.appName("s3-read-test").getOrCreate()
+data_path = "s3a://tpcds-share-emr/sf10-parquet/useDecimal=true,useDate=true,filterNull=false/call_center"
+column_name = "cc_call_center_id"
+spark.read.parquet(data_path).select(column_name).show()
+# sleep 100 seconds to allow user to check the Spark UI
```
Comment (GaryShen2008) [inline k8s/read-s3-test.py]:
```suggestion
# sleep 300 seconds to allow user to check the Spark UI
```

**Comment by GaryShen2008** [comment]
Verified the steps on AWS EKS.

File: k8s/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+ENV PATH=/opt/spark/bin:$PATH
+ENV PYTHONPATH /opt/spark/python/lib/pyspark.zip:/opt/spark/python/lib/py4j-0.10.9-src.zip:$PYTHONPATH
+
+# Install rapids plugin jar
+WORKDIR /opt/spark/jars
+RUN set -ex && \
+    wget -O rapids-4-spark-KEYS https://keys.openpgp.org/vks/v1/by-fingerprint/7A8A39909B9B202410C2A26F1D9E1285654392EF && \
+    gpg --import rapids-4-spark-KEYS && \
+    wget https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/$RAPIDS_VER/rapids-4-spark_2.12-$RAPIDS_VER-cuda11.jar.asc && \
+    wget https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/$RAPIDS_VER/rapids-4-spark_2.12-$RAPIDS_VER.jar && \
```
Comment (GaryShen2008) [inline k8s/Dockerfile]: Why download the -cuda11 jar.asc, but download a jar without -cuda11?
Let's download the jar with -cuda11.

**Comment by wjxiz1992** [comment]
@GaryShen2008 Add NOTICE file to clarify 3rd party OSS.

**Comment by wjxiz1992** [comment]
Pending internal IP/License review.

**Comment by wjxiz1992** [comment]
Add GKE support.

File: k8s/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    gpg --verify rapids-4-spark_2.12-$RAPIDS_VER-cuda11.jar.asc rapids-4-spark_2.12-$RAPIDS_VER.jar
+
+# Add necessary jars to support AWS s3 storage access
+RUN set -ex && \
+    wget https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/3.3.1/hadoop-aws-3.3.1.jar && \
```
Comment (wjxiz1992) [inline k8s/Dockerfile]: version mapping with Spark

File: k8s/Dockerfile (hunk 5 above)
Comment (wjxiz1992) [reply k8s/Dockerfile]: Done.

**Comment by wjxiz1992** [comment]
Update plugin version to 22.12 release. verified on EKS:
``` 
RAPIDS Accelerator build: {version=22.12.0, user=, url=https://github.com/NVIDIA/spark-rapids.git, date=2022-12-12T18:58:25Z, revision=6bba270bc5ce68f009c0b3e2d5b41f562cf29abb, cudf_version=22.12.0, branch=HEAD}
```

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**